include/boost/json/impl/storage_ptr.ipp

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (c) Krystian Stasiowski (sdkrystian@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/cppalliance/json
+//
+
+#ifndef BOOST_JSON_IMPL_STORAGE_PTR_IPP
+#define BOOST_JSON_IMPL_STORAGE_PTR_IPP
+
+#include <boost/json/storage_ptr.hpp>
+
+BOOST_JSON_NS_BEGIN
+
+// KRYSTIAN NOTE: we need P1077
+union no_destroy
+{
+    constexpr
+    no_destroy() noexcept
+        : resource_() 
+    {
+    }
+
+    ~no_destroy() noexcept
+    {
+    }
+
+    detail::default_resource resource_;
+};
+
+template<class T>
+struct singleton
+{
+    static no_destroy resource_;
+};
+
+template<class T>
+no_destroy singleton<T>::resource_;
```

Comment:
You are injecting all these into the `boost::json` namespace, I think they should go in `boost::json::detail`?

Reply:
I'll just move this to `default_resource`